hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/api/BooleanMatcher.kt

```diff
@@ -14,7 +16,10 @@ data class BooleanMatcher(
   @Schema(description = "Must be Boolean", example = "Boolean")
   override val type: String = "Boolean"
 
-  override fun buildQuery(): AbstractQueryBuilder<*> = QueryBuilders.termQuery(attribute, condition)
+  override fun buildQuery(attributes: Attributes): AbstractQueryBuilder<*> =
+    attributes[attribute]?.let {
+      QueryBuilders.termQuery(it.openSearchName, condition)
```

Comment:
Now all of our queries use the `openSearchName` in case it is different to the attribute name.